pkg/routes/thirdpartyrepo.go

```
@@ -124,6 +126,26 @@ func createRequest(w http.ResponseWriter, r *http.Request) (*models.ThirdPartyRe
 // GetAllThirdPartyRepo return all the ThirdPartyRepo
 func GetAllThirdPartyRepo(w http.ResponseWriter, r *http.Request) {
 	services := dependencies.ServicesFromContext(r.Context())
+	account, err := common.GetAccount(r)
+	if err != nil {
+		services.Log.WithField("error", err.Error()).Error("Error retrieving account from the request")
+		err := errors.NewBadRequest(err.Error())
+		w.WriteHeader(err.GetStatus())
+		if err := json.NewEncoder(w).Encode(&err); err != nil {
+			services.Log.WithField("error", err.Error()).Error("Error while trying to encode")
+		}
+		return
+	}
+
+	// Check to see if feature is enabled
+	unleashCtx := unleashCTX.Context{
+		UserId: account,
+	}
+	enabled := unleash.IsEnabled("fleet-management.custom-repos", unleash.WithContext(unleashCtx))
+	if !enabled {
+		respondWithJSONBody(w, services.Log, "Feature not available")
```

I think we are missing a return here ?

---

maybe we should return an error here instead of 200 , as the api contract say 200 has a special structure

---

Cool, I can change it.

My thoughts here where that perhaps an error would be confusing. An error , to me, suggests that I have done something wrong, maybe a wrong payload or bad auth, when in-fact I have done everything correctly but I just dont have access to the feature.

Happy to switch it to an error if we think thats the more appropriate solution.

---

Added an error for feature not available and gave it a status 501 :smile: